### PR TITLE
SL-619: linking existing patient to labour and delivery encounter

### PR DESCRIPTION
**Pull request by cioan**
@mogoodrich , finally I was able to complete this new feature end-to-end. Now the code checks if the selected baby is already linked to the mother and highlights that entry. Also, it allows to search and select any registered patient in the system:

Screenshot 2024-07-23 at 2 42 38 PM

### CODE REVIEW

File: omod/src/main/webapp/pages/children/children.gsp [hunk 1]
```diff
@@ -281,9 +309,13 @@ ${ ui.includeFragment("coreapps", "patientHeader", [ patient: patient ]) }
             let button = jq('#matchedPatientTemplates .local_button').clone();
             var link = patientDashboardLink;
             link += (link.indexOf('?') == -1 ? '?' : '&') + 'patientId=' + item.uuid;
-            button.attr("onclick", "linkBabyToDeliveryForm('" + registerBabyObs + "', '" + item.uuid + "')");
+            button.attr("onclick", "linkBabyToDeliveryForm('" + registerBabyObs + "', '" + item.uuid + "', '" + item.motherUuid + "')");
 
             cloned.append(button);
+            if (item.motherUuid) {
+                jq(cloned).css("background-color", "#c8dfa6");
+                cloned.find('.matchingMother').append("<p style='color: red;'>This patient is already listed as child of ${ patient.givenName } ${ patient.familyName }</p>");
```
Comment (mogoodrich) [inline omod/src/main/webapp/pages/children/children.gsp:317]: Do we need a message code here?

File: omod/src/main/webapp/pages/children/findChildren.gsp [hunk 2]
```diff
@@ -56,6 +58,27 @@
...
+                                };
+                                let dataJson = JSON.stringify(updatedObs);
+                                jq.ajax({
+                                    type: "POST",
+                                    url: apiBaseUrl + "/obs/" + laborAndDeliveryObs,
+                                    dataType: "json",
+                                    contentType: "application/json",
+                                    data: dataJson
+                                }).fail(function (data) {
+                                        emr.errorMessage("Failed to update Labor and Delivery Register baby in EMR obs: " + data.responseText);
```
Comment (mogoodrich) [inline omod/src/main/webapp/pages/children/findChildren.gsp:77]: Do these need localization?

**Review by mogoodrich — APPROVED** [review]
Generally looks good to me, though I found a few places we should probably extract out into message codes?  Thanks @cioan !

**Comment by cioan** [comment]
Thanks @mogoodrich ! I have added message codes.

**Pull request merged**